packages/media-query-mixin/media-query-mixin.ts

```diff
@@ -0,0 +1,125 @@
+import { LitElement, property } from 'lit-element';
+import { PropertyDeclaration } from 'lit-element/lib/updating-element.js';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function mediaProperty(opts?: MediaDeclaration): (proto: object, name?: PropertyKey) => any {
+  return property(opts as PropertyDeclaration);
+}
+
+export interface MediaDeclaration {
+  readonly media?: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Constructor<T = object> = new (...args: any[]) => T;
+
+export const MediaQueryMixin = <T extends Constructor<LitElement>>(
+  base: T
+): T & Constructor<LitElement> => {
+  class MediaQuery extends base {
+    private static _mediaQueries: {
+      [prop: string]: { query: MediaQueryList; updating?: boolean };
+    } = {};
+
+    private static _queryProps: Set<string> = new Set();
+
+    private _boundQueries: { [prop: string]: () => void } = {};
+
+    /**
+     * Extend the LitElement `createProperty` method to watch media query.
+     */
+    static createProperty(name: PropertyKey, options: PropertyDeclaration) {
+      const { media } = options as MediaDeclaration;
+      const prop = name as string;
+
+      const isMediaString = typeof media === 'string';
+      const opts = isMediaString
+        ? {
+            type: Boolean,
+            attribute: false,
+            noAccessor: true,
+            media
+          }
+        : options;
+
+      ((base as unknown) as typeof LitElement).createProperty(name, opts);
+
+      if (isMediaString) {
+        const query = window.matchMedia(media as string);
+        this._mediaQueries[prop] = { query };
+        this._queryProps.add(prop);
+
+        const key = `__${prop}`;
+        Object.defineProperty(this.prototype, prop, {
+          get(): boolean | null | undefined {
+            return (this as { [key: string]: boolean | null | undefined })[key];
+          },
+          set(this: MediaQuery, value: boolean | null | undefined) {
+            if (MediaQuery._mediaQueries[prop].updating !== true) {
+              return;
+            }
+            const oldValue = ((this as {}) as { [key: string]: unknown })[prop];
+            ((this as {}) as { [key: string]: unknown })[key] = value;
```

Comment:
Nit: A bit complex to understand from the first sight where we saving what values 😅 Probably needs comment with a link to https://github.com/Polymer/lit-element/blob/41e9fd3b8c18b3070f9df2a2f21915b796b2b298/src/lib/updating-element.ts#L306-L320

Reply:
Thanks, added.